lessons/specifics/iex_helpers.md

```diff
@@ -0,0 +1,250 @@
+of enumerables and provides infinite streams.
+
+Since the majority of the functions in Enum enumerate the whole enumerable and
+return a list as result, infinite streams need to be carefully used with such
+functions, as they can potentially run forever. For example:
+
+┃ Enum.each Stream.cycle([1, 2, 3]), &IO.puts(&1)
+```
+
+And now we can even combine this with the autocomplete features of our shell.
+Imagine we were exploring Map for the first time:
+
+```elixir
+iex> h Map
+                                      Map
+
+A set of functions for working with maps.
+
+Maps are key-value stores where keys can be any value and are compared using
+the match operator (===). Maps can be created with the %{} special form defined
+in the Kernel.SpecialForms module.
+
+iex> Map.
+delete/2             drop/2               equal?/2
+fetch!/2             fetch/2              from_struct/1
+get/2                get/3                get_and_update!/3
+get_and_update/3     get_lazy/3           has_key?/2
+keys/1               merge/2              merge/3
+new/0                new/1                new/2
+pop/2                pop/3                pop_lazy/3
+put/3                put_new/3            put_new_lazy/3
+split/2              take/2               to_list/1
+update!/3            update/4             values/1
+
+iex> h Map.merge/2
+                             def merge(map1, map2)
+
+Merges two maps into one.
+
+All keys in map2 will be added to map1, overriding any existing one.
+
+If you have a struct and you would like to merge a set of keys into the struct,
+do not use this function, as it would merge all keys on the right side into the
+struct, even if the key is not part of the struct. Instead, use
+Kernel.struct/2.
+
+Examples
+
+┃ iex> Map.merge(%{a: 1, b: 2}, %{a: 3, d: 4})
+┃ %{a: 3, b: 2, d: 4}
+```
+
+And as you can see, we not only found out what was offered by the module but exactly how to work what we were after with a working example!
+
+### `i`
+Since we already have another tool to learn about this and its valid Elixir code, why not use `h` to learn about `i`? Here we go:
+
+```elixir
+iex> h i
+
+                                  def i(term)
+
+Prints information about the given data type.
+
+iex> i Map
+Term
+  Map
+Data type
+  Atom
+Module bytecode
+  /usr/local/Cellar/elixir/1.3.3/bin/../lib/elixir/ebin/Elixir.Map.beam
+Source
+  /private/tmp/elixir-20160918-33925-1ki46ng/elixir-1.3.3/lib/elixir/lib/map.ex
+Version
+  [9651177287794427227743899018880159024]
+Compile time
+  no value found
+Compile options
+  [:debug_info]
+Description
+  Use h(Map) to access its documentation.
+  Call Map.module_info() to access metadata.
+Raw representation
+  :"Elixir.Map"
+Reference modules
+  Module, Atom
+```
+
+So now we have a bunch more information about `Map`, including all the way down to where its source is stored and the modules it references.
+This is quite useful when exploring custom and foreign data types and new functions.
```

Comment:
"This is quite useful when exploring custom, foreign data types, and new functions."